data_cleaning.py

```diff
@@ -0,0 +1,98 @@
+"""Clean final data."""
+
+import pandas as pd
+import geopandas as gpd
+import calendar
+import numpy as np
+
+
+def clean_data(path: str) -> pd.DataFrame:
+    """Read in data and remove nulls and outliers."""
+    df = pd.read_csv(path)
+    # Convert timestamps to datetime objects
+    df["trip_end_timestamp"] = pd.to_datetime(df["trip_end_timestamp"])
+    df["trip_start_timestamp"] = pd.to_datetime(df["trip_start_timestamp"])
+    # Drop mismatched index columns and unneeded longitude
+    df = df.drop(
+        [
+            "Unnamed: 0",
+            "Unnamed: 0.1",
+            "Unnamed: 0.1.1",
+            "pickup_centroid_location",
+            "dropoff_centroid_location",
```

Comment:
are we not using the centroid location anymore?

Reply:
Yes, good point! Because the community area shapefile has community names and numbers, I can now extract location as needed for geographic visualization by merging it with the shapefile.